### PR TITLE
Docs Use Cases section, cleanup

### LINKED ISSUE #809: Organize Use Cases in docs site
With #800 we'll have a page in the docs site that talks about a use case in the planning phase. We may want other pages to talk about current or future use cases, add diagrams, etc.

## Acceptance Criteria

- [ ] We have a section in the docs for `use-cases/`
- [ ] The new section has a brief intro page, maybe link to the roadmap?
- [ ] Students is moved to the new section
- [ ] A `Use Cases` link to the new section is available on the `Home` sidebar menu

### PR DESCRIPTION
**Pull request by thekaveman**
* Closes #809 by creating a new section for Use Cases and a brief intro page
* Simplifies the nav files and page titles
* Apply `prettier` formatting to all docs markdown files

### CODE REVIEW

File: docs/use-cases/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        height="718"
+        frameborder="0"
+        marginheight="0"
+        marginwidth="0"
+        embedded=true>
+    Loading…
+</iframe>
+
+[milestones]: https://github.com/cal-itp/benefits/milestones
+[roadmap]: https://docs.google.com/document/d/1IFoa8Ye0IXwGXXwxFjMrm1s3617Dbv6l-E-aCB0kgnA/edit#
```
Comment (afeld) [inline docs/use-cases/README.md:28]: @vykster Are you ok with us highlighting the roadmap?

**Review by afeld — DISMISSED** [review]
Pending the one question.

File: docs/use-cases/README.md (hunk 1 above)
Comment (thekaveman) [reply docs/use-cases/README.md:28]: Important to note that only folks with access to the doc can click the link / see the embed.

**Comment by thekaveman** [comment]
Sorry @afeld added one more link and rebased on your latest changes.

**Review by afeld — APPROVED** [review]

File: docs/use-cases/README.md (hunk 1 above)
Comment (thekaveman) [reply docs/use-cases/README.md:31]: Good to go per [Vyki in Slack](https://cal-itp.slack.com/archives/C037Y3UE71P/p1658355046912929?thread_ts=1658344851.168449&cid=C037Y3UE71P)

Comment (afeld) [reply docs/use-cases/README.md:31]: Merge away!

**Pull request merged**